src/game_config.cpp

```diff
@@ -313,8 +314,13 @@ void load_config(const config &v)
 	if(auto i = v.optional_child("images")){
 		using namespace game_config::images;
 
-		game_title            = i["game_title"].str();
-		game_title_background = i["game_title_background"].str();
+		// Select a background at random
+		const auto backgrounds = utils::split(i["game_title_background"].str());
+		int r = rand() % (backgrounds.size());
+		game_title_background = backgrounds.at(r);
+
+		// Allow game_title to be empty
+		game_title            = i["game_title"].str().empty() ? game_title_background : i["game_title"].str();
```

Comment:
More details? Do you mean checking with [isblank()](https://en.cppreference.com/w/cpp/string/byte/isblank)?

Reply:
this line stops a warning which shows up if `game_title` is empty.

Reply:
I mean https://github.com/wesnoth/wesnoth/blob/e67a91caf22429c56cd2012e324cd8ee212b5481/src/config_attribute_value.cpp#L357
It checks if the attribute was specified at all.

Nothing wrong with the warnings and maybe nothing wrong with ignoring them as well? I'm not suggesting that we do not set `game_title` just that it's possible to not set it. Whether that looks good or not is a different question.

Or is it fundamentally broken if `game_title` is empty?

Reply:
> Nothing wrong with the warnings and maybe nothing wrong with ignoring them as well?
If it is possible for `game_title` to be empty/blank, then isn't that warning redundant? Maybe we can just remove that warning.
Going one step further, `game_title` is simply another image that gets blitted on top of `game_title_background`, maybe that's redundant as well, since we have the IPF `~BLIT()`.

Reply:
> Going one step further, `game_title` is simply another image that gets blitted on top of `game_title_background`, maybe that's redundant as well, since we have the IPF `~BLIT()`.

Sounds like it. Though since you make it a list now then you'd have to repeat the BLIT each time? Though maybe you want to BLIT a bit differently for different images...

Reply:
> Sounds like it. Though since you make it a list now then you'd have to repeat the BLIT each time? Though maybe you want to BLIT a bit differently for different images...

No, only one image is randomly selected from the list, so you're always blitting two images.

Reply:
I think changing the condition on the warning instead of assigning a value here would be better. Instead of showing the warning when `game_title` is empty, show it instead when both `game_title` and `game_title_background` are empty (the background becomes black in this case.)

Reply:
The WML part:
https://github.com/wesnoth/wesnoth/blob/e67a91caf22429c56cd2012e324cd8ee212b5481/data/gui/window/title_screen.cfg#L34-L50

Reply:
Changed the condition for the warning, only happens when both game_title and game_background are empty.